File: src/main/java/net/alephdev/calendar/models/Risk.java
package net.alephdev.calendar.models;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.math.BigDecimal;
import lombok.Data;

@Data
@Entity
@Table(name = "Risk")
public class Risk {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Integer id;

  @Column(nullable = false)
  private String description;

  @Column(nullable = false)
  private BigDecimal probability;

  @Column(nullable = false)
  private BigDecimal estimatedLoss;
}

File: angular/src/app/models/risk.ts
/**
 * Модель риска проекта.
 *
 * Риск описывается вероятностью и оценкой потенциальных потерь.
 *
 * @property id - Уникальный идентификатор риска
 * @property description - Описание риска
 * @property probability - Вероятность наступления (0..1)
 * @property estimatedLoss - Оценка потерь при наступлении риска
 */
export interface Risk {
  id: number;
  description: string;
  probability: number;
  estimatedLoss: number;
}
